[tests] test_cache_invalidate_{by_prefix,default_clears_all} pollute on full-suite ordering — add autouse fixture

## Problem

`tests/test_api_helpers.py::TestCacheFunctions::test_cache_invalidate_by_prefix` and `test_cache_invalidate_default_clears_all` fail in the main pytest sweep on main, but PASS in isolation and in CI's fresh-process workaround invocation.

Symptom:
```
FAILED tests/test_api_helpers.py::TestCacheFunctions::test_cache_invalidate_by_prefix
FAILED tests/test_api_helpers.py::TestCacheFunctions::test_cache_invalidate_default_clears_all
```

The first asserts `dropped == 2` (after invalidating `orient_*` prefix); actual is `9` because earlier tests in the full suite populate `orient_*`-prefixed keys in the module-global `_ttl_cache`.

## Why it matters

Any PR that touches the import chain (e.g. PR #2000 `feat(audit): bulk UA-GEC F/Calque lookup table`) trips this in CI even though the PR's own code is clean. Causes false-failure noise on every unrelated PR.

## Root cause

`scripts.api.state_helpers._ttl_cache` is a module-global singleton. The two affected tests assert EXACT counts — they need the cache to start empty. CI has a workaround (`.github/workflows/ci.yml` runs them in a fresh subprocess after the main pytest), but the main sweep also runs them and that's where the failure is recorded.

## Fix

Add an autouse fixture in `TestCacheFunctions` that clears `_ttl_cache` before each test:

```python
@pytest.fixture(autouse=True)
def _clear_ttl_cache_before_each_test(self):
    from scripts.api.state_helpers import _ttl_cache
    _ttl_cache.clear()
    yield
    _ttl_cache.clear()
```

That makes the assertions robust to whatever populated the cache earlier in the full-suite run, and the CI workaround can be removed.

## Severity

Low (CI false negative on unrelated PRs), but ~30 min to fix and removes a recurring blocker.

## Related

- PR #2000 hit this in `Test (pytest)` despite the PR's code being clean
- Probably the same class as #1958 (since closed by #1995, but that fix was about template wording, not cache state — different surface)

--- a/tests/test_api_helpers.py
+++ b/tests/test_api_helpers.py
@@ -17,6 +17,8 @@ import time
 from pathlib import Path
 from unittest.mock import patch
 
+import pytest
+
 # ==================== review_parsing ====================
 
 
@@ -279,7 +281,24 @@ class TestParsePhaseStatus:
 
 
 class TestCacheFunctions:
-    """TTL cache get/set."""
+    """TTL cache get/set.
+
+    `_ttl_cache` is a module-global singleton; tests in this class assert
+    exact key counts after invalidation. To keep them robust against
+    pollution from earlier tests in the full pytest sweep (which can
+    populate `orient_*`-prefixed keys via the `/api/orient` test suite),
+    we clear the cache before AND after every test in this class.
+
+    See issue #2002 — replaces the prior CI workaround that re-ran these
+    tests in a fresh subprocess.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _clear_ttl_cache(self):
+        from scripts.api.state_helpers import _ttl_cache
+        _ttl_cache.clear()
+        yield
+        _ttl_cache.clear()
 
     def test_cache_hit(self):
         from scripts.api.state_helpers import cache_get, cache_set

